pallets/ddc-clusters/src/lib.rs

```diff
@@ -538,6 +588,44 @@ pub mod pallet {
 			Ok(())
 		}
 
+		fn do_join_cluster(
+			cluster: Cluster<T::AccountId>,
+			node_pub_key: NodePubKey,
+		) -> DispatchResult {
+			ensure!(cluster.can_manage_nodes(), Error::<T>::UnexpectedClusterStatus);
+
+			let mut node: pallet_ddc_nodes::Node<T> = T::NodeRepository::get(node_pub_key.clone())
+				.map_err(|_| Error::<T>::AttemptToAddNonExistentNode)?;
+			ensure!(node.get_cluster_id().is_none(), Error::<T>::AttemptToAddAlreadyAssignedNode);
+
+			node.set_cluster_id(Some(cluster.cluster_id));
+			T::NodeRepository::update(node).map_err(|_| Error::<T>::AttemptToAddNonExistentNode)?;
+
+			ClustersNodes::<T>::insert(
+				cluster.cluster_id,
+				node_pub_key.clone(),
+				ClusterNodeState {
+					kind: ClusterNodeKind::External,
+					status: ClusterNodeStatus::ValidationSucceeded,
```

Comment:
Do we want to treat a node that is being added to the cluster as operational/trusted from the very beginning (i.e. set `ValidationSucceeded` status immediately)? 

Ideally, it should have `AwaitsValidation` status right after it is added to the cluster, and then validated by another nodes within a cluster by some challenge. After passing (or not passing) the challenge, the status should be automatically (by validators OCWs/DAC) changed to `ValidationSucceeded` (or `ValidationFailed`).

Currently, we are missing this "operability" challenge on both DDC and blockchain sides, and that's why the validation is performed manually by sending the `validate_node` extrinsic on behalf Cluster Manager account.

Won't it be an issue if `add_node` extrinsic and `join_cluster` extrinsic lead to different node status within a cluster ? As far as I know, Cluster Manager UI utilizes the `add_node`  extrinsic for adding node to cluster on behalf cluster manager account

Reply:
> Do we want to treat a node that is being added to the cluster as operational/trusted from the very beginning (i.e. set ValidationSucceeded status immediately)?

Yes, we want it operational from the very beginning at least for now while we have no other means to check it as you mentioned. It also passed the check by the smart contract which may contain an off-chain flow with the custom connection check implemented by the cluster manager.

> Won't it be an issue if add_node extrinsic and join_cluster extrinsic lead to different node status within a cluster ?

I don't see any issue right now. Please let me know if you see any.